lfs.c

```diff
@@ -4911,3 +4920,521 @@ int lfs_migrate(lfs_t *lfs, const struct lfs_config *cfg) {
 }
 
 #endif
+
+#if LFS_THREADSAFE
+
+int lfs_format(lfs_t *lfs, const struct lfs_config *config) {
+    int err = config->lock(config);
+    if (err) {
+        return err;
+    }
+
+    err = lfs_format_raw(lfs, config);
+    config->unlock(config);
+
+    return err;
+}
+
+int lfs_mount(lfs_t *lfs, const struct lfs_config *config) {
+    int err = config->lock(config);
```

Comment:
Hmm, does it actually make sense to lock here? In theory no one else should be able to access the filesystem before it's initialized.